File: base.mvp/src/main/java/alvin/base/mvp/main/view/MainActivity.java
package alvin.base.mvp.main.view;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Button;

import alvin.base.mvp.R;
import alvin.base.mvp.android.views.AndroidActivity;
import alvin.base.mvp.basic.views.BasicActivity;
import alvin.base.mvp.multibindings.views.MultibindingsActivity;
import alvin.base.mvp.scope.views.ScopeActivity;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);

        ButterKnife.bind(this);
    }

    @OnClick({
            R.id.btn_basic,
            R.id.btn_scope,
            R.id.btn_android,
            R.id.btn_multibindings
    })
    public void onButtonsClick(Button b) {
        Intent intent;

        switch (b.getId()) {
        case R.id.btn_basic:
            intent = new Intent(this, BasicActivity.class);
            break;
        case R.id.btn_scope:
            intent = new Intent(this, ScopeActivity.class);
            break;
        case R.id.btn_android:
            intent = new Intent(this, AndroidActivity.class);
            break;
        case R.id.btn_multibindings:
            intent = new Intent(this, MultibindingsActivity.class);
            break;
        default:
            intent = null;
        }

        if (intent != null) {
            startActivity(intent);
        }
    }
}

File: ui.ime/src/main/java/alvin/ui/ime/main/views/MainActivity.kt
package alvin.ui.ime.main.views

import alvin.ui.ime.R
import alvin.ui.ime.hide.views.HideImeActivity
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.intentFor
import org.jetbrains.anko.sdk25.coroutines.onClick

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        setupButtons()
    }

    private fun setupButtons() {
        arrayOf(btn_hide_ime).forEach {
            it.onClick { btn ->
                startActivity(when (btn) {
                    btn_hide_ime -> intentFor<HideImeActivity>()
                    else -> throw IllegalArgumentException()
                })
            }
        }
    }
}
